File: PrimerC#/Backend/Backend/models/MuestraSismica.cs
namespace Backend.models
{
    public class MuestraSismica
    {
        public DateTime FechaHoraMuestra { get; set; }

        public List<DetalleMuestraSismica> Detalles { get; set; } = new List<DetalleMuestraSismica>();

        public object ObtenerDatos() 
        {
            return new
            {
                FechaHoraMuestra = FechaHoraMuestra,
                Detalles = Detalles.Select(d => new
                {
                    valor = d.Valor
                })

            };
        }
    }
}

File: PPAI Tkinter/Entitys/MuestraSismica.py
from Entitys.DetalleMuestraSismica import DetalleMuestraSismica
import datetime

class MuestraSismica:
    def __init__(self, fechaHoraMuestra: datetime, detalleMuestra : DetalleMuestraSismica):
        self._fechaHoraMuestra = fechaHoraMuestra
        self._detalles = [detalleMuestra]

# region Getters y Setters
    @property
    def fechaHoraMuestra(self):
        return self._fechaHoraMuestra
    @fechaHoraMuestra.setter
    def fechaHoraMuestra(self, valor: datetime):
        self._fechaHoraMuestra = valor

    @property
    def detalles(self):
        return self._detalles   
    @detalles.setter
    def detalles(self, valor):
        self._detalles = valor
# endregion

    def obtenerDatos(self):
        #return [detalle.obtenerDatos() for detalle in self._detalles]
        return {
            "fechaHoraMuestra": self._fechaHoraMuestra,
            "detalles": [detalle.obtenerDatos() for detalle in self._detalles]
        }

    def __repr__(self):
        return f"MuestraSismica(fechaHoraMuestra={self._fechaHoraMuestra}, detalles={self._detalles})"
    
    #TODO: metodos de implementación fuera del diagrama
    def agregarDetalle(self, detalle: DetalleMuestraSismica):
        self._detalles.append(detalle)